### PR TITLE
[ICDS Dashboard]: Bihar API Mother modifications 

### PR DESCRIPTION
**Pull request by ShubhamBansal1997**
##### SUMMARY

https://app.asana.com/0/1175081284767805/1174849920897133/f

##### PRODUCT DESCRIPTION

payload result
url = `http://localhost:8000/a/icds-cas/bihar_demographics/mother_details/?month=5&year=2017`
```json
{"data": [{"household_id": null, "ccs_case_id": "014a6b5e-97b2-47e9-9d7e-c75043a507f8", "person_id": "82b54a72-7512-4d44-93cb-449325b3021a", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": null, "preg_reg_date": null, "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-03-20", "hb": null, "lmp": "2016-11-01", "edd": "2017-08-08", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": "b6a55583-e07d-4367-ae5c-f3ff22f85271", "ccs_case_id": "08d215e7-81c7-4ad3-9c7d-1b27f0ed4bb5", "person_id": "cc75916b-a71e-4c4d-a537-5c7bef95b12f", "married": 1, "husband_name": "test_husband_name", "husband_id": "b1e7f7d8-149e-4ffc-a876-2a70a469edbc", "last_preg_year": 12, "last_preg_tt": null, "is_pregnant": 1, "preg_reg_date": "2017-04-12", "tt_1": "2017-05-01", "tt_2": "2017-05-02", "tt_booster": "2017-05-03", "add": "2017-06-01", "hb": 2, "lmp": "2016-10-02", "edd": "2017-07-09", "anc_1": "2016-10-08", "anc_2": "2016-11-07", "anc_3": "2016-12-07", "anc_4": "2017-01-06"}, {"household_id": null, "ccs_case_id": "10a53900-f65e-46b7-ae0c-f32a208c0677", "person_id": "177bdbb3-d4db-4077-9720-86a99b12ba25", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": null, "preg_reg_date": null, "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-03-20", "hb": null, "lmp": "2016-11-01", "edd": "2017-08-08", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "1744a035-56f1-4059-86f5-93fcea3c6076", "person_id": "2062b573-efb7-4079-a2ce-2d8c8efbd55a", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": null, "preg_reg_date": null, "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-03-01", "hb": null, "lmp": "2016-11-01", "edd": "2017-08-08", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "443b5bf7-ccfa-41b7-9c3a-172a8e1a41df", "person_id": "90fdf584-2696-4f51-9004-de0a9e00f1b1", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": null, "preg_reg_date": null, "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2016-12-01", "hb": null, "lmp": "2016-11-01", "edd": "2017-08-08", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "45330226-1136-40bf-8951-1b8cd7520ec1", "person_id": "f4fba3ec-4dc7-41c6-961f-5967a9daf7f9", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": null, "preg_reg_date": null, "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-03-01", "hb": null, "lmp": "2016-11-01", "edd": "2017-08-08", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "45fce010-a2cc-4645-8ed8-5c73a936496b", "person_id": "09c960fb-d027-4f7b-8058-30770c5707a6", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": null, "preg_reg_date": null, "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-06-01", "hb": null, "lmp": "2016-10-02", "edd": "2017-07-09", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "6164501b-d407-4c15-b9bc-eda507d71952", "person_id": "65e63422-d4e3-480d-81e3-ab82e84771b3", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": null, "preg_reg_date": null, "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-06-01", "hb": null, "lmp": "2016-10-02", "edd": "2017-07-09", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "70baa3c1-2c4c-4e94-85e5-27ddd2df4d29", "person_id": "13a94de0-663e-43ab-9769-15f64fa8060d", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": null, "preg_reg_date": null, "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-03-02", "hb": null, "lmp": "2016-11-01", "edd": "2017-08-08", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "7cdd0e7c-e36a-4a87-abc7-6fcff9bb109a", "person_id": "8e8aadc6-0e2f-47b9-bd92-84c50db7af9c", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": null, "preg_reg_date": null, "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-03-02", "hb": null, "lmp": "2016-11-01", "edd": "2017-08-08", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "7e13b697-39bf-4b8a-9360-ac16ce227214", "person_id": "d69550ac-30d3-4147-b134-c4a0b0b72715", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": null, "preg_reg_date": null, "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-06-02", "hb": 2, "lmp": "2016-11-01", "edd": "2017-08-08", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "99588911-ec13-4cb9-91c9-28108954599f", "person_id": "c8febe66-e622-4a99-9035-a9f7b5a7518e", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": null, "preg_reg_date": null, "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-04-02", "hb": null, "lmp": "2016-08-01", "edd": "2017-05-08", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "9bd35459-6a3c-43ad-a144-0c0013b4272e", "person_id": "8c5f8d3f-c6cd-4737-a994-2e494aeda05a", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": 1, "preg_reg_date": "2017-04-12", "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-06-01", "hb": 2, "lmp": "2016-11-24", "edd": "2017-08-31", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "a87c92ec-d5a6-4dae-a8a1-14e7abf31441", "person_id": "c013cab2-e924-43f6-9a2d-cff137774229", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": null, "preg_reg_date": null, "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-05-02", "hb": null, "lmp": "2016-10-05", "edd": "2017-07-12", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "b0e0d62b-83ab-4a61-96ad-dbe791d7eeee", "person_id": "c6bf241c-d4a4-4170-9424-5a2cfbcb79b6", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": 1, "preg_reg_date": "2017-05-12", "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-05-02", "hb": null, "lmp": "2016-11-01", "edd": "2017-08-08", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "b313b7a0-5d9f-4186-88ca-c8389211e26b", "person_id": "7b3b24ca-76e0-4fa8-97b0-cd4d6652bd76", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": null, "preg_reg_date": null, "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-06-01", "hb": null, "lmp": "2016-10-02", "edd": "2017-07-09", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "bb0798ee-2be9-4c82-ac53-0beb6a573737", "person_id": "e7f7c148-aefb-4f76-873b-33a2a5b64d59", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": null, "preg_reg_date": null, "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-03-01", "hb": null, "lmp": "2016-11-01", "edd": "2017-08-08", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "c7c368d4-e554-4e12-a62f-87403453a515", "person_id": "3c9658d5-e522-4f2c-b636-3ed194d89cb1", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": 1, "preg_reg_date": "2017-04-12", "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-06-01", "hb": null, "lmp": "2016-11-24", "edd": "2017-08-31", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "e413c896-7595-4a11-8d16-79c87cad3423", "person_id": "f76921f9-ea4f-4c4c-b6f6-de5752102e4e", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": null, "preg_reg_date": null, "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-03-01", "hb": null, "lmp": "2016-11-01", "edd": "2017-08-08", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "e67a3dd2-f6bf-4e95-9bb7-a254754a28ce", "person_id": "f36755e1-018a-448b-83b0-01a4080d6f3f", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": null, "preg_reg_date": null, "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-06-01", "hb": null, "lmp": "2016-10-02", "edd": "2017-07-09", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "f3d9aeb7-d3f5-49d1-aa78-4f33aa5cee77", "person_id": "e94393fd-5007-48f3-95fb-fc1c7f6d3bce", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": 1, "preg_reg_date": "2017-04-12", "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-06-01", "hb": null, "lmp": "2016-10-02", "edd": "2017-07-09", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "f491263a-4846-4ffd-a64d-a6653c03dd03", "person_id": "1d3e45d5-bd58-487c-9f93-7da2cf67c8d7", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": null, "preg_reg_date": null, "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-03-20", "hb": null, "lmp": "2016-11-01", "edd": "2017-08-08", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}, {"household_id": null, "ccs_case_id": "f77d2181-5850-4675-9abe-2276acca4198", "person_id": "ef12d6ce-a9af-4cf5-8459-0286432c5465", "married": null, "husband_name": null, "husband_id": null, "last_preg_year": null, "last_preg_tt": null, "is_pregnant": 1, "preg_reg_date": "2017-04-12", "tt_1": null, "tt_2": null, "tt_booster": null, "add": "2017-06-01", "hb": null, "lmp": "2016-10-02", "edd": "2017-07-09", "anc_1": null, "anc_2": null, "anc_3": null, "anc_4": null}], "metadata": {"month": 5, "year": 2017, "total_count": 23, "timestamp": "21:58:29 13 May 2020"}}
```

### CODE REVIEW

File: custom/icds_reports/tests/agg_tests/reports/test_bihar_api_mother_details.py [hunk 1]
```diff
@@ -41,7 +41,12 @@ def test_file_content(self):
                 "hb": 2,
                 "add": datetime.date(2017, 6, 1),
                 "last_preg_tt": None,
-                "lmp": datetime.date(2016, 10, 2)
+                "lmp": datetime.date(2016, 10, 2),
+                "anc_1": None,
```
Comment (Rohit25negi) [inline custom/icds_reports/tests/agg_tests/reports/test_bihar_api_mother_details.py:45]: could we update fixture to see the data here in the test to ensure this is working?

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
Gnerally looks good. Could we update the fixture to have anc values in the test(Atleast for one) to ensure its working. 
Also did you test this by curling the api that the data is coming

File: custom/icds_reports/migrations/sql_templates/database_views/bihar_api_mother_view.sql [hunk 2]
```diff
@@ -20,7 +20,12 @@ CREATE VIEW bihar_api_mother_view AS
         "ccs_record_monthly"."tt_2" AS "tt_2",
         "ccs_record_monthly"."anemia" AS "hb",
         "ccs_record_monthly"."add" AS "add",
-        "ccs_record_monthly"."edd" - 280 AS lmp
+        "ccs_record_monthly"."edd" - 280 AS lmp,
+        "ccs_record_monthly"."edd" AS edd,
+        "ccs_record_monthly"."anc_1" AS "anc_1",
+        "ccs_record_monthly"."anc_2" AS "anc_2",
+        "ccs_record_monthly"."anc_3" AS "anc_3",
+        "ccs_record_monthly"."anc_4" AS "anc_4"
```
Comment (Rohit25negi) [inline custom/icds_reports/migrations/sql_templates/database_views/bihar_api_mother_view.sql:28]: hey you also need to rebuild the view.

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
one more blocking comment

File: custom/icds_reports/tests/agg_tests/reports/test_bihar_api_mother_details.py (hunk 1 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/tests/agg_tests/reports/test_bihar_api_mother_details.py]: updated

File: custom/icds_reports/migrations/sql_templates/database_views/bihar_api_mother_view.sql (hunk 2 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/migrations/sql_templates/database_views/bihar_api_mother_view.sql:28]: updated!

**Comment by ShubhamBansal1997** [comment]
@Rohit25negi Please re-review added the sample json response in the PR description

File: custom/icds_reports/tests/agg_tests/fixtures/pregnant_tasks.csv [hunk 3]
```diff
@@ -2,4 +2,4 @@ doc_id,inserted_at,state_id,awc_id,ccs_record_case_id,tasks_type,num_anc_complet
 task_case_id1,2017-04-01,state2,a40,bfc8681e-bd2d-45f2-a811-cdffa41ddf32,pregnancy,,,2017-04-02,2017-04-07,,,,,,s8
 task_case_id2,2017-04-01,state1,a12,3a9c23af-0a21-475c-a9b8-1de4c1ed68b8,pregnancy,,,2017-03-02,,,,,,,s4
 task_case_id3,2017-04-01,state2,a39,45e84a85-9944-4027-b0eb-866a79cbe955,pregnancy,,,2017-05-02,,,,,,,s7
-task_case_id4,2017-04-01,st1,a27,08d215e7-81c7-4ad3-9c7d-1b27f0ed4bb5,pregnancy,,,,,,,2017-05-01,2017-05-02,2017-05-03,s3
+task_case_id4,2017-04-01,st1,a27,08d215e7-81c7-4ad3-9c7d-1b27f0ed4bb5,pregnancy,4,,2016-10-08,2016-11-07,2016-12-07,2017-01-06,2017-05-01,2017-05-02,2017-05-03,s3
```
Comment (Rohit25negi) [inline custom/icds_reports/tests/agg_tests/fixtures/pregnant_tasks.csv:5]: has `num_anc_complete` been changed for a reason?

**Review by Rohit25negi — APPROVED** [review]
Looks good, Left one question.

File: custom/icds_reports/tests/agg_tests/fixtures/pregnant_tasks.csv (hunk 3 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/tests/agg_tests/fixtures/pregnant_tasks.csv]: I initially feel it's related with the anc_dates but later on deep dive investigation after this it's found that even in the ccs records they are not using from the pregnant tasks they are picking from the ccs records itself so reverting back this change.
Thanks for pointing out 👍

**Pull request merged**